contracts/governance/extensions/GovernorVotesQuorumFraction.sol

```diff
@@ -107,12 +110,12 @@ abstract contract GovernorVotesQuorumFraction is GovernorVotes {
         // Make sure we keep track of the original numerator in contracts upgraded from a version without checkpoints.
         if (oldQuorumNumerator != 0 && _quorumNumeratorHistory._checkpoints.length == 0) {
             _quorumNumeratorHistory._checkpoints.push(
-                Checkpoints.Checkpoint({_blockNumber: 0, _value: SafeCast.toUint224(oldQuorumNumerator)})
+                Checkpoints.Checkpoint224({_key: 0, _value: oldQuorumNumerator.toUint224()})
             );
         }
 
         // Set new quorum for future proposals
-        _quorumNumeratorHistory.push(newQuorumNumerator);
+        _quorumNumeratorHistory.push(uint256(clock()).toUint32(), newQuorumNumerator.toUint224());
```

Comment:
The current `history` / `trace224` uses uint32 for storing the "clock". This is something we inherit from Compound:
https://github.com/compound-finance/compound-protocol/blob/master/contracts/Governance/Comp.sol#L27-L30

For block numbers is mostly ok. For timestamp (this has been discussed many times) this will overflow in `2106`. We already spent ~50 years since 1970, we have ~75 years left.

In EIP-6372, we decided to go with `uint48` for the clock ... and that is what we will use ... in 5.0
Right now we just cant change the type of the structure in case a governor wants to upgrade from <4.9 to 4.9.

NOTE: Its not just about the `updateQuorumNumerator`, its mostly about the voting token that uses the same clock.

Reply:
It's fine with both blocks and timestamps. I'm asking if there's a use case for a more granular clock since that might unexpectedly alter the behavior. Strictly talking, there's a gap between `uint32` and `uint48` that's dangerous.

Solving because of the compatibility constraint, but let's keep it in mind if there's another clock we should care about (not just Governor but other implementations).

Reply:
We agree that 32 bits is not enough for some legitimate use cases, but the current code is using 32 bits for backwards compatibility. The new EIPs (5805, 6372) have settled on 48 bits which we believe should be enough for reasonable granularity. Contracts 5.0 will be changed to use 48 bits in accordance with the EIPs.

@ernestognw I'm not sure if you're making a particular suggestion here.

I would say that we should add a warning to the contract in 4.9 that timepoints need to fit in 32 bits.

Reply:
> We agree that 32 bits is not enough for some legitimate use cases
Amazing, thanks!

> @ernestognw I'm not sure if you're making a particular suggestion here.

I'm asking because this conversation is new for me.
The warning for now sounds good to me.

Reply:
Ok let's add a warning. I suppose it should go in `GovernorVotes`?

Reply:
Hadrien also mentioned the voting tokens, so maybe every `IVotes` implementers as well